### PR TITLE
Trap for a nil pointer and give a helpful message

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

The nil pointer exceptions logged in honeybadger don't make it easy to decide what action needs to be taken.

## Was the API documentation (openapi.json) updated?
Yes

### CODE REVIEW

File: app/services/public_xml_service.rb [hunk 1]
```diff
@@ -104,6 +104,13 @@ def add_data_from_src(external_file)
     # locate and extract the resourceId/fileId elements
     doc = src_item.contentMetadata.ng_xml
     src_resource = doc.at_xpath("//resource[@id=\"#{src_resource_id}\"]")
+
+    unless src_resource
+      raise Dor::DataError, "The contentMetadata of #{object.pid} has an exernalFile "\
```
Comment (mjgiarlo) [inline app/services/public_xml_service.rb]: s/exernalFile/externalFile/

File: app/services/public_xml_service.rb [hunk 2]
```diff
@@ -104,6 +104,13 @@ def add_data_from_src(external_file)
...
+        "reference to #{src_druid}, #{src_resource_id}, but #{src_druid} doesn't have " \
+        "a matching resource node in it's contentMetadata"
```
Comment (mjgiarlo) [inline app/services/public_xml_service.rb]: s/it's/its/

File: openapi.json [hunk 3]
```diff
@@ -353,6 +353,16 @@
         "responses": {
           "200": {
             "description": "OK"
+          },
+          "422": {
+            "description": "There is a problem with the data that makes is impossible to publish"
```
Comment (mjgiarlo) [inline openapi.json]: need a comma at the end of the line?

Comment (jcoyne) [reply openapi.json]: Trailing commas are illegal in json.

Comment (jcoyne) [reply openapi.json]: oh, not trailing here.

**Pull request merged**